test/common/buffer/buffer_speed_test.cc

```diff
@@ -473,4 +474,166 @@ BENCHMARK(bufferStartsWithMatch)
+    ->Args({0, 1, 64 * 1024 * 1024, 1024, 1})
+    ->Args({0, 1, 64 * 1024 * 1024, 8192, 1})
+    ->Args({0, 1, 64 * 1024 * 1024, 1, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 8, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 16, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 32, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 64, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 128, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 1024, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 8192, 3})
+    ->Args({0, 1, 64 * 1024 * 1024, 1, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 8, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 16, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 32, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 64, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 128, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 1024, 5})
+    ->Args({0, 1, 64 * 1024 * 1024, 8192, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 1, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 8, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 16, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 32, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 64, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 128, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 1024, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 8192, 1})
+    ->Args({1, 0, 64 * 1024 * 1024, 1, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 8, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 16, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 32, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 64, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 128, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 1024, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 8192, 3})
+    ->Args({1, 0, 64 * 1024 * 1024, 1, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 8, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 16, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 32, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 64, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 128, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 1024, 5})
+    ->Args({1, 0, 64 * 1024 * 1024, 8192, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 1, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 8, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 16, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 32, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 64, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 128, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 1024, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 8192, 1})
+    ->Args({1, 1, 64 * 1024 * 1024, 1, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 8, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 16, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 32, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 64, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 128, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 1024, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 8192, 3})
+    ->Args({1, 1, 64 * 1024 * 1024, 1, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 8, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 16, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 32, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 64, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 128, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 1024, 5})
+    ->Args({1, 1, 64 * 1024 * 1024, 8192, 5});
```

Comment:
This seems like too many cases, can this be trim down while still capturing the interesting performance cases?

Reply:
I will reduce some cases.